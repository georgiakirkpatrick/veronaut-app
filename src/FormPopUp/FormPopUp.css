.FormPopUp__close-button {
    position: absolute;
    top: 0;
    right: 3vw;
    border: 0;
    color: #000;
    padding: 1vw;
    margin: 1vw;
    cursor: pointer;
    border: none;
    font: 10px;
}

.FormPopUp {
    padding: 0 5vw 15vh;
    position: fixed;
    top: 9vh;
    left: 0;
    height: 91vh;
    width: 100%;
    overflow: scroll;
    border: none;
    background-color: #fff;
    z-index: 3;
    transform: translateX(-107%);
    transition: transform 0.3s ease-out;
    overflow-y: scroll;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font: 16px/20px Poppins, sans-serif;
}

.FormPopUp .FormButton {
    font: 400 16px/22px Poppins, sans-serif;
    margin: 5vh auto;
    background-color: #F2E7F1;
}

.FormPopUp h1 {
    text-align: center;
    font: 300 30px Poppins, sans-serif;
    padding: 5vh 0;
}

.FormPopUp header {
    width: 100%;
    text-align: center;
}

.FormPopUp.active {
    transform: translateX(0);
}

.FormPopUp fieldset {
    position: relative;
    width: 100%;
    padding: 10px;
    margin: 5px 0;
    background-color: rgb(238, 238, 238);
    border: none;
}

.FormPopUp input {
    margin: 5px 0px 10px;
}

@media screen and (min-width: 768px) {
    .FormPopUp {
        padding: 0 7vw 15vh;
    }

    .FormPopUp__close-button {
        right: 5vw;
    }    
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .FormPopUp {
        top: 7vh;
        height: 93vh;
        padding: 0 10vw 15vh;
    }
    
    .FormPage h1 {
        text-align: center;
        padding: 5vh 0;
        font: 300 40px Poppins, sans-serif;
    }

    .FormPopUp input {
        margin: 5px 0 10px;
    }

    .FormPopUp__close-button {
        right: 8vw;
    }    
}

@media screen and (min-width: 1440px) {
    .FormPopUp {
        top: 10vh;
        height: 91vh;
        padding: 5vh 15vw 10vh;
    }

    .FormPopUp__close-button {
        right: 13vw;
    }
}

@media screen and (min-width: 2560px) {
    .FormPopUp {
        padding: 5vh 25vw 10vh;
    }

    .FormPopUp__close-button {
        right: 23vw;
    }
}