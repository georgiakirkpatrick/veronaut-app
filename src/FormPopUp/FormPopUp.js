import React from 'react'
import FormButton from '../FormButton/FormButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import './FormPopUp.css'

const FormPopUp = props => {
    const {
        buttonText,
        buttonType,
        children,
        close,
        id,
        status,
        submit,
        title
    } = props

    return (
        <section className={status}
            id={id}
        >
            <button
                className='FormPopUp__close-button' 
                type='button'
                onClick={close}
            >
                <FontAwesomeIcon icon='times' size='lg' />
            </button>
            
            <header>
                <h1>
                    {title}
                </h1>
            </header>
            
            {children}

            <FormButton 
                buttonText={buttonText}
                handleClick={submit}
                type={buttonType}
            />
        </section>
    )
}

FormPopUp.defaultProps = {
    buttonText: '',
    buttonType: 'button',
    children: <div />,
    close: () => {},
    id: 1,
    status: '',
    submit: () => {},
    title: ''
}

export default FormPopUp