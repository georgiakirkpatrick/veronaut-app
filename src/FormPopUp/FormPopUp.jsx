import React from 'react'
import FormButton from '../FormButton/FormButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import './FormPopUp.css'

const FormPopUp = ({
  buttonText = '',
  buttonType = 'button',
  children = <div />,
  close = () => {},
  id = 1,
  status = '',
  submit = () => {},
  title = ''
}) => {
  return (
    <section className={status}
      id={id}
    >
      <button
        className='FormPopUp__close-button' 
        type='button'
        onClick={close}
      >
      
      <FontAwesomeIcon icon='times' size='lg' />
      </button>
      
      <header>
        <h1>
          {title}
        </h1>
      </header>
      
      {children}

      <FormButton 
        buttonText={buttonText}
        handleClick={submit}
        type={buttonType}
      />
    </section>
  )
}

export default FormPopUp