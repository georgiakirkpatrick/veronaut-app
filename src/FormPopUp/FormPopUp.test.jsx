import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import FormPopUp from './FormPopUp'

describe('FormPopUp', () => {
    it ('renders without crashing', () => {
        render(<FormPopUp />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<FormPopUp />)
        expect(tree).toMatchSnapshot()
    })
})