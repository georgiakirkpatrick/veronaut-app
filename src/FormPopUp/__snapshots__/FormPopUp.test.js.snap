// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPopUp render the UI as expected 1`] = `
<section
  className=""
  id={1}
>
  <button
    className="FormPopUp__close-button"
    onClick={[Function]}
    type="button"
  />
  <header>
    <h1>
      
    </h1>
  </header>
  <div />
  <button
    className="FormButton"
    onClick={[Function]}
    type="button"
  >
    
  </button>
</section>
`;
